pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -298,21 +305,26 @@ func (a *AgentClusterInstall) finish() error {
 // Sets the default network type to OVNKubernetes if it is unspecified in the
 // AgentClusterInstall or InstallConfig
 func setNetworkType(aci *hiveext.AgentClusterInstall, installConfig *types.InstallConfig,
-	warningMessage string) {
-
+	warningMessage string) (bool, string) {
 	if aci.Spec.Networking.NetworkType != "" {
-		return
+		return false, ""
 	}
 
 	if installConfig != nil && installConfig.Networking != nil &&
 		installConfig.Networking.NetworkType != "" {
-		aci.Spec.Networking.NetworkType = installConfig.NetworkType
-		return
+		if installConfig.Networking.NetworkType == string(operv1.NetworkTypeOVNKubernetes) || installConfig.Networking.NetworkType == string(operv1.NetworkTypeOpenShiftSDN) {
+			aci.Spec.Networking.NetworkType = installConfig.NetworkType
+			return false, ""
+		}
+
+		aci.Spec.Networking.NetworkType = string(operv1.NetworkTypeOVNKubernetes)
+		return true, installConfig.NetworkType
```

Comment:
This is a bit confusing because `installConfig.NetworkType` and `installConfig.Networking.NetworkType` are actually the same field.
I don't think we need to return this here anyway though.

Reply:
Yeah I overcomplicated that a bit, fixed.